bee-ledger/src/workers/consensus/worker.rs

```diff
@@ -28,15 +29,31 @@ use bee_tangle::{ConflictReason, MsTangle, TangleWorker};
 
 use async_trait::async_trait;
 
-use futures::stream::StreamExt;
+use futures::{channel::oneshot, stream::StreamExt};
 use log::{error, info, warn};
 use tokio::sync::mpsc;
 use tokio_stream::wrappers::UnboundedReceiverStream;
 
 use std::{any::TypeId, convert::TryInto};
 
 /// Event of the consensus worker.
-pub struct ConsensusWorkerEvent(pub MessageId);
+#[allow(clippy::type_complexity)]
+pub enum ConsensusWorkerEvent {
+    /// Event to confirm a milestone.
+    Confirm(MessageId),
+    /// Event to fetch the balance of an address.
+    Balance(Address, oneshot::Sender<(Result<Option<Balance>, Error>, LedgerIndex)>),
+    /// Event to fetch an output.
+    Output(
+        OutputId,
+        oneshot::Sender<(Result<Option<CreatedOutput>, Error>, LedgerIndex)>,
+    ),
+    /// Event to fetch the outputs of an address.
+    Outputs(
+        Address,
+        oneshot::Sender<(Result<Option<Vec<OutputId>>, Error>, LedgerIndex)>,
+    ),
+}
```

Comment:
Do we want to put this type into its own module. At some point we may want to add an additional `impl` block to this event type containing some convenience methods, and also impl certain traits for it. Wdyt?

Also: probably typo in doc comment: Event**s**.

Reply:
Two additional suggestions:
From my understanding of the code those events have a `command`-like notion to them. Events usually don't prescribe what the listener has to do upon receiving them. A command however has this notion. So what I would have done here (just mentioning my viewpoint here!) is to call it `ConsensusWorkerCommand` to indicate what the receiver is supposed to do, and also include the instruction as an imperative to the enum variants, so:
`ConfirmMilestone`, `FetchBalance`, `FetchOutput`, `FetchOutputs`.

Reply:
I changed to Command and the new names. Regarding the 1st comment, I think this is consistent with the rest of the codebase like that, I think it's easier to follow if the Event/Command type is together with the worker that consumes them.

Reply:
alright 👌